File: C++/String.cpp
#include <stdio.h>
#include <iostream>
#include <regex>

using namespace std;

int main(void) {

    // C++はstringクラスがある
    string str2 = "Honda";
    printf("%s \n", str2.c_str());

    // 結合
    str2 += "Yamaha";
    printf("%s \n", str2.c_str());

    // 更新
    str2 = "Kawasaki";
    printf("%s \n", str2.c_str());

    // 文字列長
    printf("length=%d \n", str2.length());

    // 比較
    if (str2 == "Kawasaki") {
        printf("str2 == Kawasaki \n");
    }

    // 置換
    str2 = "ABCDEFG";
    str2.replace(0,2, "XX"); // 開始位置, 長さ, 置換する文字列
    printf("%s \n", str2.c_str());
    string str3 = regex_replace(str2, regex("FG"), "ZZ"); // 対象の文字列, regex(正規表現を指定),置換する文字
    printf("%s \n", str3.c_str());


    return 0;
}

File: Ruby/String.rb
# 出力
str = "ABCDE"
p str
puts str
print str + "\n"

# 変数展開
print("文字列は#{str}\n")

# 連結
puts str + "EFG"

# 繰り返し
puts str * 3

# 文字列長
str = "あいうえお"
puts str.length
puts str.size

# 文字列の中の特定の文字数
str = "あいうえおあいう"
puts str.count("い")

# 比較
str = "あいうえお"
str2 = "あいうえお"
if str == str2 then
    puts "same"
end
if "ABC".casecmp("abC") then
    puts "same caseInsensitive"
end

# 分割
str = "Honda, Yamaha, Kawasaki"
ary = str.split(",")
p ary

# スライス
str = "ABCDEFG"
puts str.slice(1, 3) # 2番目から3文字分

# 置換
str = "ABCDE"
puts str.tr("ABC", "ZZZ")

# 大文字小文字
puts "ABC".upcase
puts "abc".downcase
puts "abc".capitalize

# trim
str = "  ab cd  "
puts str.strip
puts str.lstrip # 先頭のみ
puts str.rstrip # 末尾のみ

# 改行を取り除く
str = " ab cd \n"
puts str.chomp

# 文字列、数値の変換
puts "1234".to_i
puts 1234.to_s
